scripts/tybalt_vae.py

```diff
@@ -0,0 +1,320 @@
+import keras
+
+import pydot
+import graphviz
+from keras.utils import plot_model
+from IPython.display import SVG
+from keras.utils.vis_utils import model_to_dot
+
+
+# In[2]:
+
+print(keras.__version__)
+tf.__version__
+
+
+# ## Load Functions and Classes
+# 
+# This will facilitate connections between layers and also custom hyperparameters
+
+# In[3]:
+
+# Function for reparameterization trick to make model differentiable
+def sampling(args):
+    
+    import tensorflow as tf
+    # Function with args required for Keras Lambda function
+    z_mean, z_log_var = args
+
+    # Draw epsilon of the same shape from a standard normal distribution
+    epsilon = K.random_normal(shape=tf.shape(z_mean), mean=0.,
+                              stddev=epsilon_std)
+    
+    # The latent vector is non-deterministic and differentiable
+    # in respect to z_mean and z_log_var
+    z = z_mean + K.exp(z_log_var / 2) * epsilon
+    return z
+
+
+class CustomVariationalLayer(Layer):
+    """
+    Define a custom layer that learns and performs the training
+
+    """
+    def __init__(self, **kwargs):
+        # https://keras.io/layers/writing-your-own-keras-layers/
+        self.is_placeholder = True
+        super(CustomVariationalLayer, self).__init__(**kwargs)
+
+    def vae_loss(self, x_input, x_decoded):
+        reconstruction_loss = original_dim * metrics.binary_crossentropy(x_input, x_decoded)
+        kl_loss = - 0.5 * K.sum(1 + z_log_var_encoded - K.square(z_mean_encoded) - 
+                                K.exp(z_log_var_encoded), axis=-1)
+        return K.mean(reconstruction_loss + (K.get_value(beta) * kl_loss))
+
+    def call(self, inputs):
+        x = inputs[0]
+        x_decoded = inputs[1]
+        loss = self.vae_loss(x, x_decoded)
+        self.add_loss(loss, inputs=inputs)
+        # We won't actually use the output.
+        return x
+
+
+# ### Implementing Warm-up as described in Sonderby et al. LVAE
+# 
+# This is modified code from https://github.com/fchollet/keras/issues/2595
+
+# In[4]:
+
+class WarmUpCallback(Callback):
+    def __init__(self, beta, kappa):
+        self.beta = beta
+        self.kappa = kappa
+    # Behavior on each epoch
+    def on_epoch_end(self, epoch, logs={}):
+        if K.get_value(self.beta) <= 1:
+            K.set_value(self.beta, K.get_value(self.beta) + self.kappa)
+
+
+# In[5]:
+
+get_ipython().magic('matplotlib inline')
+plt.style.use('seaborn-notebook')
+
+
+# In[6]:
+
+np.random.seed(123)
+
+
+# ## Load Gene Expression Data
+
+# In[7]:
+
+rnaseq_file = os.path.join('data', 'pancan_scaled_zeroone_rnaseq.tsv')
+rnaseq_df = pd.read_table(rnaseq_file, index_col=0)
+print(rnaseq_df.shape)
+rnaseq_df.head(2)
+
+
+# In[8]:
+
+# Split 10% test set randomly
+test_set_percent = 0.1
+rnaseq_test_df = rnaseq_df.sample(frac=test_set_percent)
+rnaseq_train_df = rnaseq_df.drop(rnaseq_test_df.index)
+
+
+# ## Initialize variables and hyperparameters
+
+# In[9]:
+
+# Set hyper parameters
+original_dim = 5000
```

Comment:
Hyperparameter question: Is the original dim data-matrix dependent? i.e. if the data-matrix were to have a shape greater than 5000 would this need to change?